Changelog: Tunnel hostname routing is available to all Cloudflare One users (#25271)
cloudflare/cloudflare-docs: src/content/changelog/cloudflare-tunnel/2025-09-18-tunnel-hostname-routing.mdx

Previously, Tunnel routes could only be defined by IP address or [CIDR range](/cloudflare-one/connections/connect-networks/private-net/cloudflared/connect-cidr/). This created a challenge for modern applications with dynamic or ephemeral IP addresses, often forcing administrators to maintain complex and brittle IP lists.

![Hostname-based routing in Cloudflare Tunnel](~/assets/images/changelog/cloudflare-one/tunnel-hostname-routing.webp)

**What’s new:**
- **Hostname & Domain Routing**: Create routes for individual hostnames (e.g., `payroll.acme.local`) or entire domains (e.g., `*.acme.local`) and direct their traffic to a specific Tunnel.
- **Simplified Zero Trust Policies**: Build resilient policies in Cloudflare Access and Gateway using stable hostnames, making it dramatically easier to apply per-resource authorization for your private applications.
- **Precise Egress Control**: Route traffic for public hostnames (e.g., `bank.example.com`) through a specific Tunnel to enforce a dedicated source IP, solving the IP allowlist problem for third-party services.
- **No More IP Lists**: This feature makes the workaround of maintaining dynamic IP Lists for Tunnel connections obsolete.

> Get started in the Tunnels section of the Zero Trust dashboard with your first [hostname route](/cloudflare-one/connections/connect-networks/private-net/cloudflared/connect-private-hostname/) and connect your [private networks](/cloudflare-one/connections/connect-networks/private-net/cloudflared/) with [Cloudflare Tunnel](/cloudflare-one/connections/connect-networks/).

Reviewer: ```suggestion
Get started in the Tunnels section of the Zero Trust dashboard with your first [private hostname](/cloudflare-one/connections/connect-networks/private-net/cloudflared/connect-private-hostname/) or [public hostname](/cloudflare-one/policies/gateway/egress-policies/egress-cloudflared/) route.
```
